src/components/CustomTableHead/CustomTableHead.tsx

```diff
@@ -0,0 +1,111 @@
+import { Box, Checkbox, TableCell, TableHead, TableRow, TableSortLabel } from '@mui/material';
+import { ChangeEvent, MouseEvent } from 'react';
+import { CustomTableProps } from '../CustomTable/CustomTable.types';
+import { SortingDirection } from '../../common/utils/SortingDirection';
+import { t } from 'i18next';
+
+export function CustomTableHead<T extends Record<PropertyKey, unknown>>(props: CustomTableProps<T>) {
+  const { columnConfig, selectedRowsIds, rows, setColumnConfig, setSelectedRowsIds, tableName } = props;
+
+  const isSelected = (row: T) => {
+    return (
+      selectedRowsIds.filter((item) => {
+        return (row.id as string) === item;
+      }).length > 0
+    );
+  };
+
+  const createSortHandler = (property: keyof T) => {
+    return (event: MouseEvent<unknown>) => {
+      handleRequestSort(event, property);
+    };
+  };
+
+  function updateSortingRule(index: number, columnName: string, sortDirection: SortingDirection) {
+    const updatedConfig = [...columnConfig];
+    updatedConfig.splice(index, 1, { id: columnName as keyof T, label: columnName.toUpperCase(), sortDirection: sortDirection });
```

Comment:
i think `id` !== `columnName` if you send such request to backend, it will do not know what metric should be sorted

Reply:
changed the variable name from columnName to keyName

Reply:
should be fixed in [efaf6df](https://github.com/dorm-social-room-manager/dsrm-frontend/pull/38/commits/efaf6df4eba0d295d5d24b95bba84cc3faca74e2)